frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -948,6 +887,128 @@ function ReaderBookmark:renameBookmark(item, from_highlight)
+                                if not self:isBookmarkMatchTable(bm_menu.item_table[i], match_table) then
+                                    table.remove(bm_menu.item_table, i)
+                                end
+                            end
+                            bm_menu:switchItemTable(_("Bookmarks (search results)"), bm_menu.item_table)
+                            self.filtered_mode = true
+                        else -- from main menu
+                            self:onShowBookmark(match_table)
+                        end
+                    end,
+                },
+            },
+        },
+    }
+    check_button_case = CheckButton:new{
+        text = " " .. _("Case sensitive"),
+        checked = false,
+        parent = input_dialog,
+        max_width = input_dialog._input_widget.width,
+        callback = function()
+            check_button_case:toggleCheck()
+        end,
+    }
+    input_dialog:addWidget(check_button_case)
+    separator = CenterContainer:new{
+        dimen = Geom:new{
+            w = input_dialog._input_widget.width,
+            h = 2 * Size.span.vertical_large,
+        },
+        LineWidget:new{
+            background = Blitbuffer.COLOR_DARK_GRAY,
+            dimen = Geom:new{
+                w = input_dialog._input_widget.width,
+                h = Size.line.medium,
+            }
+        },
+    }
+    input_dialog:addWidget(separator)
+    check_button_highlight = CheckButton:new{
+        text = " " .. DISPLAY_PREFIX["highlight"] .. _("highlights"),
+        checked = true,
+        parent = input_dialog,
+        max_width = input_dialog._input_widget.width,
+        callback = function()
+            check_button_highlight:toggleCheck()
+        end,
+    }
+    input_dialog:addWidget(check_button_highlight)
+    check_button_note = CheckButton:new{
+        text = " " .. DISPLAY_PREFIX["note"] .. _("notes"),
+        checked = true,
+        parent = input_dialog,
+        max_width = input_dialog._input_widget.width,
+        callback = function()
+            check_button_note:toggleCheck()
+        end,
+    }
+    input_dialog:addWidget(check_button_note)
+    check_button_bookmark = CheckButton:new{
+        text = " " .. DISPLAY_PREFIX["bookmark"] .. _("page bookmarks"),
+        checked = true,
+        parent = input_dialog,
+        max_width = input_dialog._input_widget.width,
+        callback = function()
+            check_button_bookmark:toggleCheck()
+        end,
+    }
+    input_dialog:addWidget(check_button_bookmark)
+
+    UIManager:show(input_dialog)
+    input_dialog:onShowKeyboard()
+end
+
+function ReaderBookmark:isBookmarkMatchTable(item, match_table)
+    if match_table[item.type] then
+        if match_table.search_str == "" then
+            return true
+        else
+            local text = item.notes
+            if item.text then -- search in the highlighted text and in the note
+                text = text .. "|" .. item.text
```

Comment:
This `|` feels not that rare :) It can be inputed on the keyboard, and found in books.
Just use `"\u{FFFF}"`